tests/components/ecobee/test_climate.py

```diff
@@ -1,288 +1,585 @@
+            "desiredHeat": 753,
+            "desiredCool": 734,
+            "desiredFanMode": "auto",
+        },
+        "events": [
+            {"type": "hold", "name": "Home", "running": True, "holdClimateRef": "home"},
+            {
+                "type": "template",
+                "name": "_Default_",
+                "running": False,
+                "holdClimateRef": "",
+            },
+        ],
+        "program": {
+            "climates": [
+                {"name": "Home", "climateRef": "home"},
+                {"name": "Away", "climateRef": "away"},
+                {"name": "Sleep", "climateRef": "sleep"},
+            ],
+            "currentClimateRef": "home",
+        },
+        "equipmentStatus": "humidifier",
+    },
+    {
+        "identifier": "test-identifier-5",
+        "name": "Test Thermostat 5",
+        "modelNumber": "athenaSmart",
+        "settings": {
+            "hvacMode": "cool",
+            "coolStages": 0,
+            "heatStages": 0,
+            "fanMinOnTime": 5,
+            "heatCoolMinDelta": 50,
+            "holdAction": "nextTransition",
+        },
+        "runtime": {
+            "connected": True,
+            "actualTemperature": 727,
+            "actualHumidity": 35,
+            "desiredHeat": 753,
+            "desiredCool": 734,
+            "desiredFanMode": "auto",
+        },
+        "events": [
+            {
+                "type": "sensor",
+                "name": "Living Room",
+                "running": True,
+                "holdClimateRef": "",
+            },
+            {
+                "type": "template",
+                "name": "_Default_",
+                "running": False,
+                "holdClimateRef": "",
+            },
+        ],
+        "program": {
+            "climates": [
+                {"name": "Home", "climateRef": "home"},
+                {"name": "Away", "climateRef": "away"},
+                {"name": "Sleep", "climateRef": "sleep"},
+            ],
+            "currentClimateRef": "sleep",
+        },
+        "equipmentStatus": "auxHeat2,fan",
+    },
+]
+
+
+@pytest.fixture
+def mock_account(hass):
+    """Mock an ecobee account object."""
+    account = ecobee.EcobeeData(
+        hass,
+        MockConfigEntry(domain=ecobee.DOMAIN),
+        api_key="dummy",
+        refresh_token="dummy",
+    )
+
+    def mock_get_thermostat(index):
+        return account.ecobee.thermostats[index]
+
+    account.ecobee = Mock(
+        thermostats=MOCK_API_RESPONSE,
+        get_thermostat=mock_get_thermostat,
+        spec=pyecobee.Ecobee,
+    )
+
+    return account
+
+
+async def setup_climate(hass, mock_account):
+    """Load the ecobee climate platform with the mocked account."""
+    hass.config.components.add(ecobee.DOMAIN)
+    hass.data[ecobee.DOMAIN] = mock_account
+    config_entry = MockConfigEntry(domain=ecobee.DOMAIN)
+    await hass.config_entries.async_forward_entry_setup(config_entry, "climate")
```

Comment:
Set up the entry instead. That will set up the ecobee component too and then we don't need to add test code that mocks that. We need to patch the ecobee library class to avoid I/O.

https://github.com/home-assistant/home-assistant/blob/d63ea198f2a9d65e2be4160f33688534744605df/homeassistant/config_entries.py#L666